### PR TITLE
calendar-shared-types

### PR DESCRIPTION
**Pull request by JoshuaGoldberg**
## Changes

Shared types for calendar improvement

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: src/shared/src/types/calendar-types.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  startTime?: Date;
+  endTime?: Date;
+  recurrenceNumber: number;
+  initialDateScheduled: Date;
+  allDay: Boolean;
+}
+
+export interface EventType {
+  name: string;
+  events: Event[];
```
Comment (wavehassman) [inline src/shared/src/types/calendar-types.ts]: I think we might want the event to have the specific event type ID instead of the event type have the array

**Review by wavehassman — CHANGES_REQUESTED** [review]
that is my only thought, trying to think ahead on when we'll need the array or ID in the future
